pandas/core/generic.py

```diff
@@ -192,10 +192,7 @@ class NDFrame(PandasObject, SelectionMixin):
     _data = None  # type: BlockManager
 
     if TYPE_CHECKING:
-        # TODO(PY36): replace with _attrs : Dict[Hashable, Any]
-        # We need the TYPE_CHECKING, because _attrs is not a class attribute
-        # and Py35 doesn't support the new syntax.
-        _attrs = {}  # type: Dict[Optional[Hashable], Any]
+        _attrs = {}  # type: Dict[Hashable, Any]
```

Comment:
I don't think the optional should be removed. This looks like an inconsistency between the code and the TODO.

Reply:
and the intent was to remove the TYPE_CHECKING block and use py3.6 syntax for the variable annotation